go/nbs/store.go

```diff
@@ -183,41 +183,27 @@ func (nbs *NomsBlockStore) GetMany(hashes []hash.Hash) []chunks.Chunk {
 		return
 	}()
 
-	sort.Sort(getRecordByPrefix(reqs))
-
 	if remaining {
-		tables.getMany(reqs, wg)
+		tables.getMany(reqs, foundChunks, wg)
 		wg.Wait()
 	}
-
-	sort.Sort(getRecordByOrder(reqs))
-
-	resp := make([]chunks.Chunk, len(hashes))
-	for i, req := range reqs {
-		if req.data == nil {
-			resp[i] = chunks.EmptyChunk
-		} else {
-			resp[i] = chunks.NewChunkWithHash(hashes[i], req.data)
-		}
-	}
-
-	return resp
 }
 
-func toGetRecords(hashes []hash.Hash) []getRecord {
+func toGetRecords(hashes hash.HashSet) []getRecord {
 	reqs := make([]getRecord, len(hashes))
-	for i, h := range hashes {
+	for h, _ := range hashes {
 		a := addr(h)
-		reqs[i] = getRecord{
+		reqs = append(reqs, getRecord{
```

Comment:
Since you're appending here, instead of setting elements directly by index, I believe you need to set the capacity of `reqs` to 0 above. Otherwise, you'll wind up with a slice containing `len(hashes)` empty records, and then all the stuff you care about appended to that.